sources/VT100Screen.m

```diff
@@ -2712,6 +2727,9 @@ - (void)terminalCarriageReturn {
         [currentGrid_ moveCursorToLeftMargin];
     }
     [delegate_ screenTriggerableChangeDidOccur];
+    if (commandStartX_ != -1) {
+        [delegate_ screenCommandDidEndWithRange:[self commandRange]];
```

Comment:
I don't think this is a good idea. `screenCommandDidEndWithRange:` should only be called when you're totally sure the command ended (you got an escape sequence saying so) or the situation is hopeless (the screen resized). A shell might use carriage return when redrawing a prompt. It would be safer to call `screenCommandDidChangeWithRange:` here.